python/runtime/tensorflow/input_fn.py

```diff
@@ -143,12 +172,30 @@ def read_feature_as_tensor(raw_val, feature_spec, feature_name):
     if feature_spec["delimiter"] == "":
         return [raw_val]
     if feature_spec["is_sparse"]:
-        indices = tf.strings.to_number(
-            tf.strings.split(raw_val,
-                             feature_spec["delimiter"],
-                             result_type='RaggedTensor'), tf.int64)
-        values = tf.fill(tf.shape(indices), 1)
-        indices = tf.expand_dims(indices, 1)
+        if feature_spec["delimiter2"] != "":
+            kvlist = tf.strings.split(raw_val,
+                                      feature_spec["delimiter"],
+                                      result_type='RaggedTensor')
+            kvsplited = tf.strings.split(
+                kvlist, feature_spec["delimiter2"],
+                result_type='RaggedTensor').to_tensor()
+            # slice key tensor and value tensor
+            indices = tf.reshape(tf.slice(kvsplited, [0, 0], [-1, 1]), [-1])
+            indices = tf.expand_dims(
+                tf.strings.to_number(indices, feature_spec["dtype"]), 1)
+            values = tf.cond(
+                tf.equal(tf.shape(kvsplited)[1],
+                         2), lambda: tf.strings.to_number(
```

Comment:
Can you explain when there would be `tf.shape(kvsplitted)[1] != 2` if `feature_spec["delimiter2"] != ""`?